contracts/interfaces/ISharedConsumer.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {IRequestTypeBase} from "./IRequestTypeBase.sol";
+
+interface ISharedConsumer {
```

Comment:
Let's keep single source of truth. If this contract contains all types to make it more explicit, then let SharedConsumer.sol extend it. If not just keep external function for other contracts to call.